web/src/features/PreflightChecks/api/getPreflightResult.tsx

```diff
@@ -0,0 +1,122 @@
+import { useQuery } from "react-query";
+import { Utilities } from "@src/utilities/utilities";
+import { PreflightCheck, PreflightResponse } from "../types";
+import { useState } from "react";
+
+async function getPreflightResult({
+  apiEndpoint = process.env.API_ENDPOINT,
+  slug,
+  sequence,
+}: {
+  apiEndpoint?: string;
+  slug: string;
+  sequence?: number;
+}): Promise<PreflightResponse> {
+  const getUrl = sequence ?
+    `${apiEndpoint}/app/${slug}/sequence/${sequence}/preflight/result` :
+    `${apiEndpoint}/app/${slug}/preflight/result`;
+  const jsonResponse = await fetch(
+    getUrl,
+    {
+      method: "GET",
+      headers: {
+        Authorization: Utilities.getToken(),
+      },
+    }
+  );
+
+  if (!jsonResponse.ok) {
+    throw new Error(`Encountered an error while fetching preflight results: Unexpected status code: ${jsonResponse.status}`);
+  }
+
+  const response: PreflightResponse = await jsonResponse.json();
+
+  // unmarshall these nested JSON strings
+  if (typeof response?.preflightResult?.result === "string")
+    response.preflightResult.result =
+      JSON.parse(response.preflightResult.result);
+
+  if (typeof response?.preflightProgress === "string")
+    response.preflightProgress = JSON.parse(response.preflightProgress);
+
+  return response;
+}
+
+function flattenPreflightResponse({ refetchCount, response }:
+  { refetchCount: number, response: PreflightResponse, }): PreflightCheck {
+
+  if (typeof response?.preflightProgress === "string" || typeof response?.preflightResult?.result === "string")
+    throw new Error("Preflight response is not properly unmarshalled");
+
+  return {
+    // flatten the error strings out into an array
+    errors: response?.preflightResult?.result
+      ?.errors?.map(error => error.error) || [],
+    pendingPreflightCheckName: response?.preflightProgress?.currentName || "",
+    // TODO: see if we can calculate a real %
+    pendingPreflightChecksPercentage: refetchCount === 0 ? 0 :
+      refetchCount > 21 ? 96 : (refetchCount * 4.5),
+    preflightResults: response?.preflightResult?.result?.results
+      ?.map(responseResult => ({
+        learnMoreUri: responseResult.uri || "",
+        message: responseResult.message || "",
+        title: responseResult.title || "",
+        showCannotFail: responseResult?.strict || false,
+        showFail: responseResult?.isFail || false,
+        showPass: responseResult?.isPass || false,
+        showWarn: responseResult?.isWarn || false,
+      })) || [],
+    showDeploymentBlocked: response?.preflightResult?.result?.results
+      ?.find(result => result?.isFail && result?.strict) ? true : false,
```

Comment:
Decided to elevate this so it's easier to handle in the components as a simple boolean instead of forcing the components to search every result on every render. 

https://github.com/replicatedhq/kots/blob/ea5b1e5bed2ac1fdfc6459cb8f6bf604cc8bebc1/web/src/features/PreflightChecks/types.ts#L34-L43

https://github.com/replicatedhq/kots/blob/fbc20947791ce613b30b68b9074292ea0bdcd5aa/web/src/components/PreflightRenderer.jsx#L84-L88